### PR TITLE
[Ready] Perfect forwarding and some forgotten stuff

### PR DESCRIPTION
**Pull request by ShFil119**
I forgot about this branch. ;)
#edit, now it should it be mergeable.

### CODE REVIEW

File: rwengine/src/TextureArchive.cpp [hunk 1]
```diff
@@ -46,7 +46,7 @@ std::unique_ptr<TextureArchive> TextureArchive::create(
                    bufSize);
         }
 
-        textureArchive->textures.push_back(std::move(texture));
+        textureArchive->textures.push_back(texture);
```
Comment (husho) [inline rwengine/src/TextureArchive.cpp]: Why remove ```std::move``` here

File: rwengine/src/loaders/LoaderIDE.cpp [hunk 2]
```diff
@@ -237,7 +237,7 @@ bool LoaderIDE::load(const std::string &filename, const PedStatsList &stats) {
                         getline(buffstream, buff, ',');
                         node.other_thing2 = atoi(buff.c_str());
 
-                        path.nodes.push_back(std::move(node));
+                        path.nodes.push_back(node);
```
Comment (husho) [inline rwengine/src/loaders/LoaderIDE.cpp]: Why remove ```std::move``` here

File: rwengine/src/TextureArchive.cpp (hunk 1 above)
Comment (ShFil119) [reply rwengine/src/TextureArchive.cpp]: Trivially copyable.

File: rwengine/src/loaders/LoaderIDE.cpp (hunk 2 above)
Comment (ShFil119) [reply rwengine/src/loaders/LoaderIDE.cpp]: Also trivially copyable.

**Pull request merged**